ts/twitter/src/app/home/_components/timeline/timeline-feed-service.ts

```diff
@@ -0,0 +1,80 @@
+import { ERROR_MESSAGES } from "@/lib/constants/error-messages";
+import { Post } from "@/lib/models/post";
+import { FakeTimelineFeedService } from "./__fake__/fake-timeline-feed-service";
+
+export type TimelineEventResponse =
+  | TimelineAccessedResponse
+  | PostCreatedResponse
+  | PostDeletedResponse;
+
+export type TimelineEventType = "TimelineAccessed" | "PostCreated" | "PostDeleted";
+
+interface TimelineAccessedResponse {
+  event_type: TimelineEventType & "TimelineAccessed";
+  posts: Post[];
+}
+
+interface PostCreatedResponse {
+  event_type: TimelineEventType & "PostCreated";
+  posts: Post[];
+}
+
+interface PostDeletedResponse {
+  event_type: TimelineEventType & "PostDeleted";
+  posts: Post[];
+}
```

Comment:
```suggestion
interface TimelineAccessedResponse {
  event_type: "TimelineAccessed";
  posts: Post[];
}

interface PostCreatedResponse {
  event_type: "PostCreated";
  posts: Post[];
}

interface PostDeletedResponse {
  event_type: "PostDeleted";
  posts: Post[];
}
```
I've reconsidered. Using an intersection type to specify `event_type` doesn't seem ideal. Instead, let's remove `TimelineEventType` and directly assign the string literal. TypeScript can handle type narrowing without `TimelineEventType`.